frontend/app/blueprints/public/static/js/basket.js

```diff
@@ -0,0 +1,136 @@
+// functions relating to the sample basket and navbar
+async function openGrapeTree({newick, sampleIds, clusterMethod}) {
+    // Open grape tree
+    const form = document.getElementById("open-tree-form")
+    // add metadata and newick file to form
+    const newickInput = form.querySelector("#newick-content")
+    newickInput.setAttribute("value", newick)
+    const metadataInput = form.querySelector("#metadata-content")
+    const typingDataInput = form.querySelector("#typing-data-content")
+    typingDataInput.setAttribute("value", clusterMethod)
+    metadataInput.setAttribute("value", JSON.stringify({sample_id: sampleIds}))
+    const submitBnt = form.querySelector("input[type=submit]")
+    submitBnt.click()
+}
+
+// cluter all samples in basket
+async function clusterSamplesInBasket(element) {
```

Comment:
What about a more expressive name for `element`, i.e. `baskedElement` (if it is a basket)